packages/tooltip/src/index.tsx

```diff
@@ -285,14 +290,74 @@ function useTooltip<T extends HTMLElement>({
     return () => ownerDocument.removeEventListener("keydown", listener);
   }, []);
 
-  function handleMouseEnter() {
+  React.useEffect(() => {
+    /*
+      This is a workaround for using tooltips with disabled controls in Safari.
+      Safari fires `pointerenter` but does not fire `pointerleave`
+      and `onPointerEventLeave` added to the trigger element will not work
+    */
+
+    if (!("PointerEvent" in window)) return;
+
+    let ownerDocument = getOwnerDocument(ownRef.current)!;
+
+    function listener(event: MouseEvent) {
+      // @ts-ignore `disabled` does not exist on HTMLDivElement but tooltips can be used with different elements
+      if (state !== VISIBLE || !ownRef.current?.disabled) return;
+
+      let target = event.target as Element | null;
+      if (
+        target?.hasAttribute("data-reach-tooltip-trigger") ||
+        target?.closest("[data-reach-tooltip-trigger][aria-describedby]")
+      ) {
+        return;
+      }
+
+      transition(GLOBAL_MOUSE_MOVE);
+    }
+
+    ownerDocument.addEventListener("mousemove", listener);
+    return () => ownerDocument.removeEventListener("mousemove", listener);
+  }, []);
+
+  function wrapMouseEvent<EventType extends React.SyntheticEvent | Event>(
+    theirHandler: ((event: EventType) => any) | undefined,
+    ourHandler: (event: EventType) => any
+  ) {
+    // Use internal MouseEvent handler only if PointerEvent not supported
+    if ("PointerEvent" in window) return theirHandler;
+
+    return wrapEvent(theirHandler, ourHandler);
+  }
+
+  function wrapPointerEventHandler(
+    handler: (event: React.PointerEvent) => any
+  ) {
+    return function onPointerEvent(event: React.PointerEvent) {
+      // Handle pointer events only from mouse device
+      if (event.pointerType !== POINTER_TYPE_MOUSE) return;
+      handler(event);
+    };
+  }
+
+  function handleMouseOrPointerEnter() {
     transition(MOUSE_ENTER, { id });
   }
 
-  function handleMouseMove() {
+  function handleMouseOrPointerMove() {
```

Comment:
Nit here, but can we keep all of these function names `handleMouse` instead of `handleMouseOrPointer`? I recognize why you changed these, but considering we're still only acting on mouse events I think the intent is better communicated as-is.

Reply:
✅  Updated